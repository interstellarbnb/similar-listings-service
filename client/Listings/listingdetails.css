.buttoncancel {
  margin: 5px;
  cursor: pointer;
  position: relative;
  display: inline-block;
  text-align: center;
  text-decoration: none;
  border-radius: 4px;
  width: auto;
  font-size: 19px;
  line-height: 22px;
  padding-top: 12px;
  padding-bottom: 12px;
  color: #484848;
  font-weight: 700;
  border-width: 2px;
  border-style: solid;
  padding-left: 24px;
  padding-right: 24px;
  min-width: 77.66563145999496px;
  background: transparent;
  border-color: #008489;
}

.buttoncreate {
  margin: 5px;
  cursor: pointer;
  position: relative;
  display: inline-block;
  text-align: center;
  text-decoration: none;
  border-radius: 4px;
  width: auto;
  font-size: 19px;
  line-height: 22px;
  padding-top: 12px;
  padding-bottom: 12px;
  color: #ffffff;
  font-weight: 700;
  border-width: 2px;
  border-style: solid;
  padding-left: 24px;
  padding-right: 24px;
  min-width: 77.66563145999496px;
  background: #008489;
  border-color: #008489;
}

.inputcreate {
  font-size: 19px;
  line-height: 24px;
  color: #484848;
  font-weight: 300;
  border-width: 1px;
  background-color: transparent;
  border-color: #aaa;
  padding: 11px;
  width: 100%;
}

.container {
    display: flex;
    height: 128px;
    align-content: space-around;
    font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto, Helvetica Neue, sans-serif;
}

.header {
  margin: 20px;
  height: 60px;
}

.item {
  margin: 20px;
}

.list {
    margin-top: 16px;
    margin-bottom: 16px;
    word-wrap: break-word;
    font-size: 19px;
    height: 400px;
    line-height: 24px;
    padding-top: 0px;
    padding-bottom: 0px;
    font-weight: 300;
  }

  .listitem {
    margin: 20px;
  }

  .listname {
    width: 90%;
    display: inline-block;;
  }
  .listheart {
    width: 10%;
    display: inline-block;
  }

  .line {
    border-bottom: 1px solid #DBDBDB;
    margin: 20px;
  }

.title {
    font-weight: normal;
    margin: 0px;
    word-wrap: break-word;
    font-size: 19px;
    line-height: 24px;
    padding-top: 0px;
    padding-bottom: 0px;
}

.location {
    margin: 0px;
    word-wrap: break-word;
    font-size: 15px;
    line-height: 18px;
    letter-spacing: 0.2px;
    padding-top: 5px;
    padding-bottom: 5px;
    font-weight: 300;
}

.review {
  font-weight: normal;
  margin: 0px;
  word-wrap: break-word;
  font-size: 15px;
  line-height: 18px;
  letter-spacing: 0.2px;
  padding-top: 5px;
  padding-bottom: 5px;
  display: inline;
}
