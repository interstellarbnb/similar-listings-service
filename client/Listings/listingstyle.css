.container {
    display: flex;
    height: 300px;
    align-content: space-around;
}

.title {
  margin-left: 80px;
  font-weight: bold;
  font-size: 25px;
}
