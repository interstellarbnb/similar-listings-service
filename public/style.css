body {
    font-family: Circular,"Helvetica Neue",Helvetica,Arial,sans-serif;
    font-size: 14px;
    line-height: 1.43;
    color: #484848;
    background-color: #fff;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}
