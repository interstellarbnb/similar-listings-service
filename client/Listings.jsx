import React, { Component } from 'react';
import axios from 'axios';
import { Carousel } from 'react-bootstrap';
import style from './listingstyle.css';
import ListingEntry from './ListingEntry';

class Listings extends Component {
  constructor(props) {
    super(props);
    this.state = {
      listings: [],
      listingId: Math.floor(Math.random() * 101),
      index: 0,
      direction: null,
    };
    this.handleSelect = this.handleSelect.bind(this);
  }

  componentDidMount() {
    this.getSimilarListings();
  }

  getSimilarListings() {
    axios.get(`/listings/${this.state.listingId}`).then((response) => {
      this.setState({ listings: response.data });
    }).catch((err) => {
      if (err) {
        throw err;
      }
    });
  }

  handleSelect(selectedIndex, e) {
    // alert(`selected=${selectedIndex}, direction=${e.direction}`);
    this.setState({
      index: selectedIndex,
      direction: e.direction
    });
  }
  render() {
    return (
      <div className={style.container}>
        <Carousel
          activeIndex={this.state.index}
          direction={this.state.direction}
          onSelect={this.handleSelect}
        >
          {this.state.listings.map(listing => (
            <Carousel.Item key={listing.id}>
              <ListingEntry listing={listing} key={listing.id} class={listing.id} />
            </Carousel.Item>
        ))
      }
        </Carousel>
      </div>
    );
  }
}

export default Listings;
