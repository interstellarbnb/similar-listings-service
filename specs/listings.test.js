import React from 'react';
import { shallow, mount, render } from 'enzyme';
import axiosMockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import style from './listingstyle.css';
import Listings from '../client/Listings/Listings';

describe('<Listings />', () => {
  let wrapper;
  const axiosMock = new axiosMockAdapter(axios);

  const listing = {
    id: 1,
    title: 'Refined Black Hole',
    price: 300,
    imageUrl: 'https://source.unsplash.com/333x222/?crater',
    reviews: [3, 5, 1],
    avgRating: 3,
    type: 'Alpha Base',
    bedCount: 1,
    city: 'San Francisco',
    state: 'CA',
    country: 'USA',
  };

  axiosMock.onGet('/listings/1').reply(404);
  axiosMock.onGet('/listings/2').reply(200, listing);

  beforeEach(() => {
    wrapper = shallow(<Listings />);
  });

  test('should render component', (done) => {
    expect(wrapper).toMatchSnapshot();
    done();
  });

  test('should change isModalOpen to false when closeModal is called', (done) => {
    wrapper.instance().closeModal();
    expect(wrapper.instance().state.isModalOpen).toBe(false);
    done();
  });

  test('should render modal when a listing is passed in', (done) => {

    wrapper.instance().openModal(listing);
    expect(wrapper.instance().state.isModalOpen).toBeTruthy();
    expect(wrapper.find('#modal').length).toBe(1);
    done();
  });

  test('should not render carousel if listings array is empty', (done) => {
    wrapper.instance().state.listings = [];
    expect(wrapper.find(`.${style.container}`).text()).toBe('Loading...');
    done();
  });

  test('should call renderCarousel method', (done) => {
    wrapper.instance().renderCarousel = jest.fn();
    wrapper.instance().setState({ listings: [listing, listing, listing] });
    expect(wrapper.instance().renderCarousel).toHaveBeenCalled();
    done();
  });

  test('should update listings upon GET request', (done) => {
    wrapper.instance().setState({ listingId: 2 });
    wrapper.instance().getSimilarListings().then(() => expect(wrapper.instance().state.listings).toEqual(listing));
    done();
  });

  test('should handle error if GET request fails', (done) => {
    wrapper.instance().setState({ listingId: 1 });
    wrapper.instance().getSimilarListings().then(response => expect(response).toThrow());
    done();
  });
});
