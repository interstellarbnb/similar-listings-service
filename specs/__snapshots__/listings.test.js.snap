// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Listings /> should render component 1`] = `
<div>
  <div
    className="title"
  >
    Similar Listings
  </div>
  <div
    className="container"
  >
    Loading...
  </div>
  <div
    id="modal"
  />
</div>
`;
